frontend/message_context.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react'
+import { Snackbar } from '@material-ui/core';
+import MuiAlert from '@material-ui/lab/Alert';
+
+/**
+ * The message context allows other complenents to display user messages in a
+ * consistent manner.
+ */
+
+export type MessageLevel = 'error' | 'warning' | 'info' | 'success';
+
+export interface MessageContext {
+    userMessage(msg: string, level: MessageLevel): void,
+}
+const emptyContext: MessageContext = {
+    userMessage() {},
+}
+
+export const MessageContext = React.createContext(emptyContext)
+
+/**
+ * Exports the context variables and functions to be used by other components
+ */
+export function useMessageContext(): MessageContext {
+  return React.useContext(MessageContext)
+}
+
+export function WithMessageContext(props: {children: React.ReactNode}) {
+  const [severity, setSeverity] = React.useState('info' as MessageLevel)
+  const [message, setMessage] = React.useState('' as string)
+  const [open, setOpen] = React.useState(false);
+
+  function userMessage(message: string, severity: MessageLevel) {
+      setMessage(message);
+      setSeverity(severity);
+      setOpen(true);
+  }
+
+  const handleClose = (event: any, reason: string) => {
+      if (reason === 'clickaway') {
```

Comment:
Enum would be better here I think.

Reply:
`reason` is sent (as a string) from the ui library not created by us